src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/ChangeLogIssueKeyExtractor.java

```diff
@@ -18,11 +26,21 @@
  */
 public final class ChangeLogIssueKeyExtractor implements IssueKeyExtractor {
 
+    private static final Logger log = LoggerFactory.getLogger(ChangeLogIssueKeyExtractor.class);
+
     public Set<String> extractIssueKeys(final WorkflowRun workflowRun) {
 
         final Set<IssueKey> allIssueKeys = new HashSet<>();
         final List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets =
-                workflowRun.getChangeSets();
+                new ArrayList<>(workflowRun.getChangeSets());
+
+        WorkflowRun previous = workflowRun.getPreviousBuild();
+        while (Objects.nonNull(previous) && !isBuildSuccessful(previous)) {
+            log.info("Extract issue keys from build: " + previous.getNumber());
+
```

Comment:
@srnjak, May I know the business  benefits/outcome for this change?

Reply:
This log gives you insight into the last successful build and all failed builds before it.
This info is useful for us to find out, which builds the changesets came from.

If you think it is not necessary, this log can be removed or just changed to a debug level.

Reply:
Okies. Change looks ok for me. I will approve the change and will do a round of testing and release it.